csrc/mmdeploy/net/ipu/ipu_net.cpp

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) OpenMMLab. All rights reserved.
+
+#include "ipu_net.h"
+
+#include <sys/time.h>
+#include <unistd.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "mmdeploy/core/logger.h"
+#include "mmdeploy/core/model.h"
+#include "mmdeploy/core/utils/formatter.h"
+
+namespace mmdeploy::framework {
+
+mmdeploy::DataType IPUNet::ipu_type_convert(const popef::DataType& ipu_type) {
+  mmdeploy::DataType mtype;
+  if (ipu_type == popef::DataType::F32) {
+    mtype = mmdeploy::DataType::kFLOAT;
+  } else if (ipu_type == popef::DataType::F16) {
+    mtype = mmdeploy::DataType::kHALF;
+  } else if (ipu_type == popef::DataType::F8) {
+    mtype = mmdeploy::DataType::kINT8;
+  } else {
+    throw std::invalid_argument(
+        "invalid data type for IPU backend, current legit is : fp32, fp16, fp8");
+  }
+  return mtype;
+}
+
+void IPUNet::copy_output(const model_runtime::TensorMemory& from, Tensor& to) {
+  if (from.data_size_bytes != to.byte_size()) {
+    MMDEPLOY_ERROR("output tensor size not match from size {} to size {}", from.data_size_bytes,
+                   to.byte_size());
+    return;
+  }
+  int size = from.data_size_bytes;
+
+  MMDEPLOY_INFO("copy output total byte_size {}", size);
+
+  char* from_ptr = static_cast<char*>(from.data.get());
+
+  char* pto = to.data<char>();
+  for (int i = 0; i < size; i++) {
+    // to->data[i] = typeid(*to).name()(pfrom[i]);  // hidden type conversion here
+    pto[i] = *(from_ptr + i);
+  }
+}
+
+void IPUNet::copy_input(const Tensor& from, model_runtime::TensorMemory& to) {
+  if (from.byte_size() != to.data_size_bytes) {
+    MMDEPLOY_ERROR("input tensor size not match  from size {} to size {} ", from.byte_size(),
+                   to.data_size_bytes);
+    return;
+  }
+  int size = from.byte_size();
+
+  MMDEPLOY_INFO("copy input total byte_size {}", size);
```

Comment:
Since the func `copy_input` is called in the forward func, it would be better to use `MMDEPLOY_DEBUG` instead.